### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by lheagy**
- mappings can take nP or a mesh
- meshIO

- data misfit looks at survey for std, eps
- FDEM.py uses u instead of f for fields
- added 1D FDEM inversion example  
- use survey.makeSyntheticData for 1D FDEM, TDEM examples
- Zero has a transpose
- mu_0, epsilon_0 accessible from EM (EM.mu_0, EM.epsilon_0)

### CODE REVIEW

File: SimPEG/Utils/matutils.py [hunk 1]
```diff
@@ -37,6 +40,9 @@ def mkvc(x, numDims=1):
 
 def sdiag(h):
     """Sparse diagonal matrix"""
+    if isinstance(h, Zero):
+        return h
```
Comment (lheagy) [inline SimPEG/Utils/matutils.py]: this should return a new instance of Zero(), not h?

Comment (rowanc1) [inline SimPEG/Utils/matutils.py]: I guess it only matters if zero has a shape to it, which I don't think we have yet..?

Comment (lheagy) [inline SimPEG/Utils/matutils.py]: right, it purposely doesn't have a shape, but is this potentially confusing to read?

File: SimPEG/Examples/EM_FDEM_1D_Inversion.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
+from SimPEG import *
```
Comment (grosenkj) [inline SimPEG/Examples/EM_FDEM_1D_Inversion.py]: We shouldn't do this. Much rather
```python
import SimPEG [as something]
or
from SimPEG import Mesh, Maps, ...etc
```
Makes things easier to follow.

**Comment by grosenkj** [comment]
I think this looks good. Just made a general comment on the import of SimPEG in examples, see Issue #235

**Pull request merged**